app/frontend/src/features/profile/view/leaveReference/formSteps/submit/SubmitHostRequestReference.tsx

```diff
@@ -0,0 +1,84 @@
+import { useMediaQuery, useTheme } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import { REFERENCE_SUCCESS, SUBMIT } from "features/profile/constants";
+import { useWriteHostReference } from "features/profile/hooks/referencesHooks";
+import ReferenceOverview from "features/profile/view/leaveReference/formSteps/submit/ReferenceOverview";
+import { useReferenceData } from "features/profile/view/leaveReference/ReferenceDataContext";
+import {
+  ReferenceFormProps,
+  useReferenceStyles,
+} from "features/profile/view/leaveReference/ReferenceForm";
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import { useParams } from "react-router";
+import { WriteHostRequestReferenceInput } from "service/references";
+
+export default function SubmitHostRequestReference({
```

Comment:
Find a way to remove this file completely and combine with the other one, just branch to decide which mutation to use. (You will have to include both hooks before branching, though).